File: app/src/main/java/com/kirchhoff/example/githubclient/repository/keyvalue/KeyValueStorage.java
package com.kirchhoff.example.githubclient.repository.keyvalue;

import android.support.annotation.NonNull;

public interface KeyValueStorage {

    String TOKEN_KEY = "TOKEN_KEY";
    String USER_NAME_KEY = "USER_NAME_KEY";

    void saveToken(@NonNull String token);

    @NonNull
    String getToken();

    void saveUserName(@NonNull String userName);

    @NonNull
    String getUserName();

    void clear();
}

File: app/src/main/java/com/kirchhoff/example/githubclient/repository/keyvalue/KeyValueStorage.kt
package com.kirchhoff.example.githubclient.repository.keyvalue


/**
 * @author Kirchhoff-
 */

interface KeyValueStorage {

    companion object {
        val TOKEN_KEY = "TOKEN_KEY"
        val USER_NAME_KEY = "USER_NAME_KEY"
    }

    fun saveToken(token: String)

    val token: String

    fun saveUserName(userName: String)

    val userName: String

    fun clear()
}
